x/evm/keeper/state_transition.go

```diff
@@ -392,10 +420,20 @@ func (k *Keeper) ApplyMessageWithConfig(
 		// - reset sender's nonce to msg.Nonce() before calling evm.
 		// - increase sender's nonce by one no matter the result.
 		stateDB.SetNonce(sender.Address(), msg.Nonce)
-		ret, _, leftoverGas, vmErr = evm.Create(sender, msg.Data, leftoverGas, msg.Value)
+
+		var reply CreateReply
+		vmErr = sgxRpcClient.Call("SgxServer.Create", args, &reply)
```

Comment:
@amaury1093 The param `args` should contain evm (sgx side) handle ID that will be interacted.

Reply:
See https://github.com/Inco-fhevm/ethermint/pull/2#discussion_r1543080522.

However, thinking more about it, it might be a good idea to still have some kind of ID per EVM instance. The ID can be the TxHash also.